identity-model/src/IdentityModel/Client/Messages/TokenIntrospectionResponse.cs

```diff
@@ -19,45 +19,67 @@ public class TokenIntrospectionResponse : ProtocolResponse
     /// <returns></returns>
     protected override Task InitializeAsync(object? initializationData = null)
     {
-        if (!IsError)
+        if (IsError)
         {
-            if (Json == null)
+            return Task.CompletedTask;
+        }
+
+        var contentType = HttpResponse?.Content?.Headers.ContentType?.MediaType;
+        if (contentType is $"application/{JwtClaimTypes.JwtTypes.IntrospectionJwtResponse}" && !string.IsNullOrWhiteSpace(Raw))
+        {
+            var parts = Raw!.Split('.');
+            if (parts.Length != 3)
             {
-                throw new InvalidOperationException("Json is null"); // TODO better exception
+                throw new InvalidOperationException("Invalid JWT format");
             }
-            var issuer = Json?.TryGetString("iss");
-            var claims = Json?.ToClaims(issuer, "scope").ToList() ?? new List<Claim>();
 
-            // due to a bug in identityserver - we need to be able to deal with the scope list both in array as well as space-separated list format
-            var scope = Json?.TryGetValue("scope");
+            var payload = parts[1];
+            var jsonString = Base64Url.Decode(payload);
+            using var rootDoc = JsonDocument.Parse(jsonString);
 
```

Comment:
We should be validating the signature, typ, iss, and aud. See section 5 of RFC 9701 for the rules for what the AS must do. We should also validate that the iat is reasonably recent.